## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 443966`
Commit: [`6f59c7eb0d823a40acd62296389e85b1ba046395`](https://github.com/NixOS/nixpkgs/commit/6f59c7eb0d823a40acd62296389e85b1ba046395) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/6f59c7eb0d823a40acd62296389e85b1ba046395..pull/443966/head))
Merge: [`2ed2effb2506c7be0ffd19acb200d985bf6763be`](https://github.com/NixOS/nixpkgs/commit/2ed2effb2506c7be0ffd19acb200d985bf6763be)

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/17821769210


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>gnunet</li>
    <li>gnunet-gtk</li>
    <li>gnunet-messenger-cli</li>
    <li>libgnunetchat</li>
    <li>taler-challenger</li>
    <li>taler-exchange</li>
    <li>taler-mdb</li>
    <li>taler-merchant</li>
    <li>taler-sync</li>
    <li>taler-twister</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>gnunet</li>
    <li>gnunet-gtk</li>
    <li>gnunet-messenger-cli</li>
    <li>libgnunetchat</li>
    <li>taler-challenger</li>
    <li>taler-exchange</li>
    <li>taler-mdb</li>
    <li>taler-merchant</li>
    <li>taler-sync</li>
    <li>taler-twister</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>gnunet</li>
    <li>gnunet-gtk</li>
    <li>gnunet-messenger-cli</li>
    <li>libgnunetchat</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>gnunet</summary>
<pre>      | ^
../src/lib/testing/testing_api_cmd_exec.c:217:1: error: conflicting types for &#x27;GNUNET_TESTING_cmd_exec_va&#x27;
  217 | GNUNET_TESTING_cmd_exec_va (
      | ^
../src/include/gnunet_testing_lib.h:697:1: note: previous declaration is here
  697 | GNUNET_TESTING_cmd_exec_va (
      | ^
2 errors generated.
[76/795] Compiling C object src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_finish.c.o
FAILED: [code=1] src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_finish.c.o 
clang -Isrc/lib/testing/libgnunettesting.3.dylib.p -Isrc/lib/testing -I../src/lib/testing -Isrc/include -I../src/include -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -D_GNU_SOURCE -DGNUNET_GTK_EXTRA_LOGGING=0 -DGETTEXT_PACKAGE=gnunet -DHAVE_CONFIG_H -fno-strict-aliasing -Wno-address-of-packed-member -Wno-tautological-constant-out-of-range-compare -DDARWIN -D_APPLE_C_SOURCE -D__APPLE_USE_RFC_3542 -fno-common -MD -MQ src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_finish.c.o -MF src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_finish.c.o.d -o src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_finish.c.o -c ../src/lib/testing/testing_api_cmd_finish.c
../src/lib/testing/testing_api_cmd_finish.c:188:1: error: conflicting types for &#x27;GNUNET_TESTING_cmd_finish&#x27;
  188 | GNUNET_TESTING_cmd_finish (
      | ^
../src/include/gnunet_testing_lib.h:644:1: note: previous declaration is here
  644 | GNUNET_TESTING_cmd_finish (
      | ^
1 error generated.
[77/795] Generating symbol file src/lib/util/libgnunetutil.18.dylib.p/libgnunetutil.18.dylib.symbols
ninja: build stopped: subcommand failed.</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>gnunet</li>
    <li>gnunet-gtk</li>
    <li>gnunet-messenger-cli</li>
    <li>libgnunetchat</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>gnunet</summary>
<pre>      | ^
1 error generated.
[77/795] Compiling C object src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_exec.c.o
FAILED: [code=1] src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_exec.c.o 
clang -Isrc/lib/testing/libgnunettesting.3.dylib.p -Isrc/lib/testing -I../src/lib/testing -Isrc/include -I../src/include -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -D_GNU_SOURCE -DGNUNET_GTK_EXTRA_LOGGING=0 -DGETTEXT_PACKAGE=gnunet -DHAVE_CONFIG_H -fno-strict-aliasing -Wno-address-of-packed-member -Wno-tautological-constant-out-of-range-compare -DDARWIN -D_APPLE_C_SOURCE -D__APPLE_USE_RFC_3542 -fno-common -MD -MQ src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_exec.c.o -MF src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_exec.c.o.d -o src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_exec.c.o -c ../src/lib/testing/testing_api_cmd_exec.c
../src/lib/testing/testing_api_cmd_exec.c:187:1: error: conflicting types for &#x27;GNUNET_TESTING_cmd_exec&#x27;
  187 | GNUNET_TESTING_cmd_exec (
      | ^
../src/include/gnunet_testing_lib.h:684:1: note: previous declaration is here
  684 | GNUNET_TESTING_cmd_exec (
      | ^
../src/lib/testing/testing_api_cmd_exec.c:217:1: error: conflicting types for &#x27;GNUNET_TESTING_cmd_exec_va&#x27;
  217 | GNUNET_TESTING_cmd_exec_va (
      | ^
../src/include/gnunet_testing_lib.h:697:1: note: previous declaration is here
  697 | GNUNET_TESTING_cmd_exec_va (
      | ^
2 errors generated.
[78/795] Compiling C object src/lib/testing/libgnunettesting.3.dylib.p/testing_api_cmd_get_topo.c.o
ninja: build stopped: subcommand failed.</pre>
</details>
</details>

--- a/pkgs/by-name/gn/gnunet/package.nix
+++ b/pkgs/by-name/gn/gnunet/package.nix
@@ -39,11 +39,11 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "gnunet";
-  version = "0.24.3";
+  version = "0.25.0";
 
   src = fetchurl {
     url = "mirror://gnu/gnunet/gnunet-${finalAttrs.version}.tar.gz";
-    hash = "sha256-WwaJew6ESJu7Q4J47HPkNiRCsuBaY+QAI+wdDMzGxXY=";
+    hash = "sha256-LepmLuhgWUaFKvAtKAbKZP2t7cxxju72uG4LJoIsNv8=";
   };
 
   enableParallelBuilding = true;
@@ -94,6 +94,12 @@ stdenv.mkDerivation (finalAttrs: {
     # builds.
     find . \( -iname \*test\*.c -or -name \*.conf \) | \
       xargs sed -i -e "s|/tmp|$TMPDIR|g"
+
+    # fix error: conflicting types for `GNUNET_TESTING_cmd_{exec,finish}`
+    for name in exec finish; do
+      substituteInPlace src/lib/testing/testing_api_cmd_$name.c \
+        --replace-fail 'const struct GNUNET_TESTING_Command' 'struct GNUNET_TESTING_Command'
+    done
   '';
 
   # unfortunately, there's still a few failures with impure tests

--- a/pkgs/by-name/li/libgnunetchat/package.nix
+++ b/pkgs/by-name/li/libgnunetchat/package.nix
@@ -16,12 +16,12 @@
 
 stdenv.mkDerivation (finalAttrs: {
   name = "libgnunetchat";
-  version = "0.5.3";
+  version = "0.6.0";
 
   src = fetchgit {
     url = "https://git.gnunet.org/libgnunetchat.git";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-DhXPYa8ya9cEbwa4btQTrpjfoTGhzBInWXXH4gmDAQw=";
+    hash = "sha256-pRO8i7tHynCqm97RLMBOiWKCl2CAYBE6RXfyIljIiQ0=";
   };
 
   strictDeps = true;
@@ -51,6 +51,7 @@ stdenv.mkDerivation (finalAttrs: {
     pkgConfigModules = [ "gnunetchat" ];
     description = "Library for secure, decentralized chat using GNUnet network services";
     homepage = "https://git.gnunet.org/libgnunetchat.git";
+    changelog = "https://git.gnunet.org/libgnunetchat.git/plain/ChangeLog?h=v${finalAttrs.version}";
     license = lib.licenses.gpl3Plus;
     platforms = lib.platforms.all;
     teams = with lib.teams; [ ngi ];

